### PR TITLE
[Sln hack share do not merge/full review yet] properly register ordered serialization

### PR DESCRIPTION
**Pull request by MansurAshraf**
@ianoc Let me know if you want me to revert import ordering. Stupid IDE formats that on commit

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/package.scala [hunk 1]
```diff
@@ -73,4 +73,6 @@ package object scalding {
       !filter.accept(p)
     }
   }
+
+  val state = new java.util.concurrent.ConcurrentHashMap[String, String]()
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/package.scala]: 0) this looks really scary and architecturally bad news. I hope we can find some other way.

1) this should be `private[scalding]` at the very least.

2) can we add plenty of comments here as to what the use case is here.

File: scalding-core/src/main/scala/com/twitter/scalding/Job.scala [hunk 2]
```diff
@@ -214,7 +210,14 @@ class Job(val args: Args) extends FieldConversions with java.io.Serializable {
 
   private def executionContext: Try[ExecutionContext] =
     Config.tryFrom(config).map { conf =>
-      ExecutionContext.newContext(conf)(flowDef, mode)
+      val updatedConfig = if (conf.getRequireOrderedSerialization) {
+        state.asScala.foldLeft(conf) {
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/Job.scala]: why only add these when we require ordered serialization? We want to get the benefit even when we don't require all of them to be ordered. Some of them can still be a win.

Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/Job.scala]: ExecutionApp does not hit this code path. We should go deeper and look inside ExecutionContext.run to get to the shared code path.

**Comment by johnynek** [comment]
We can close this now that #1293 is merged right?

**Comment by ianoc** [comment]
Think so

**Pull request closed without merging**